plugins/router/arris-sb6183

```diff
@@ -0,0 +1,366 @@
+#!/usr/bin/env python3
+
+# Copyright 2020 Nathaniel Clark <nathaniel.clark@misrule.us>
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Library General Public License as published by
+# the Free Software Foundation; version 2 only
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Library General Public License for more details.
+#
+# You should have received a copy of the GNU Library General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.
+#
+
+"""
+=head1 NAME
+
+arris-sb6183 - Health monitoring plugin for Arris SB6183 Cable Modem
+
+=head1 DESCRIPTION
+
+This provides the following multigraphs:
+* upstream and downstream power levels
+* downstream signal to noise ratio
+* downstream error counts
+* uptime
+
+The values are retrieved from the cable modem's status web pages at
+192.168.100.1. So, this plugin must be installed on a munin node
+which can access those pages.
+
+=head1 CONFIGURATION
+
+Make sure 192.168.100.1 is accessible through your firewall.
+
+To have this register with munin as it's own host set the "env.hostname" in config.
+Also ensure that the hostname set is listed in munin.conf.
+
+[arris*]
+env.hostname modem
+
+=head1 TESTING
+
+Developed and tested with:
+firmware:         D30CM-OSPREY-2.4.0.1-GA-02-NOSH
+hardware version: 1
+
+=head1 VERSION
+
+0.0.1
+
+=head1 AUTHOR
+
+Nathaniel Clark <nathaniel.clark@misrule.us>
+
+=head1 LICENSE
+
+GPLv2
+
+=head1 MAGIC MARKERS
+
+ #%# family=contrib
+ #%# capabilities=autoconf
+
+=cut
+"""
+
+import re
+import os
+import sys
+from urllib import request
+
+HOSTNAME = os.getenv("hostname", None)
+STATUS_URL = "http://192.168.100.1/RgConnect.asp"
```

Comment:
Are you sure, that the IP address of the router is always the same? Or maybe make it configurable?

(I leave this up to you, since you know that device ...)

Reply:
The documentation lists this as default and every router I've had from Arris has had this IP.  (There is no way *I* can change the IP, maybe my cable company can)

Reply:
The 192.168.100.1 seems to be industry standard.

Reply:
A local IP that cannot be changed by the local administrator/user. That sounds horrible :(

Anyway: in this case it is not necessary to make it configurable.